web/dashboard/src/data/events/actions.ts

```diff
@@ -0,0 +1,98 @@
+'use server';
+import { z } from 'zod';
+import { db } from '@/db';
+import {
+  eventsSchema,
+  EventWithDiscriminatedUnions,
+  eventWithDiscriminatedUnions,
+  NewEvent,
+} from '@/data/events/dto';
+import { events, eventsTypeEnum } from '@/data/events/schema';
+import { actionError, actionSuccess, actionZodError } from '@/data/utils';
+import { eq, sql } from 'drizzle-orm';
+import { users } from '@/data/users/schema';
+import { components, componentVersions } from '@/data/components/schema';
+import { ActionResponse } from '@/data/action';
+import { ProjectWithOwners } from '@/data/projects/dto';
+import { projects } from '@/data/projects/schema';
+import { componentVersionsSchema } from '@/data/components/dto';
+import { members } from '@/data/member/schema';
+import { alias } from 'drizzle-orm/pg-core';
+
+
+export const insertEvent = async (event: NewEvent) => {
+  try {
+    const eventInserted = await db.insert(events).values(event).returning();
+    const eventInsertedParsed = eventsSchema.safeParse(eventInserted[0]);
+    console.debug('Event inserted into database.', eventInsertedParsed);
+  } catch (error) {
+    console.error('Failed to insert event into database.', error);
+  }
+};
+
+export async function getResourceEvents(
+  resourceId: string,
+): ActionResponse<EventWithDiscriminatedUnions[]> {
+
+  if (!resourceId) {
+    return actionError('No resource id provided.');
+  }
+
+  try {
+    const members = alias(users, 'members');
+    const select = await db
+      .select({
+        ...events,
+        user: users,
+        data: {
+          version: componentVersions,
+          project: projects,
+          component: components,
+          member: members,
+          permissions:  sql`(${events.data}->>'permissions')::text`,
+        },
+      })
+      .from(events)
+      .leftJoin(users, eq(events.user_id, users.id))
+      .leftJoin(projects, sql`(${events.data}->>'project_id')
+                              ::uuid  =
+                              ${projects.id}`)
+      .leftJoin(componentVersions, sql`(${events.data}->>'version_id')::uuid =  ${componentVersions.id}`)
+      .leftJoin(components, sql`(${events.data}->>'component_id')::uuid = ${components.id}`)
+      .leftJoin(members, sql`(${events.data}->>'member_id')::text = ${members.id}`)
```

Comment:
Can you please explain the usage of member in this query? I fail to see how it is a relavant join as it is a table that only stores resource and user id relations, and both of the are already present in the table. 
The rest of the data is not relevant since it is temporal data, the permissions and roles can change across time and the state at the time of the event should be recorded

Reply:
We need this to show the username when someone changes their permission on the project.